packages/block-library/src/navigation-menu/block-colors-selector.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { IconButton, Dropdown, Toolbar } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { DOWN } from '@wordpress/keycodes';
+import { ColorPaletteControl, ContrastChecker } from '@wordpress/block-editor';
+
+/**
+ * Color Selector Icon component.
+ * @return {*} React Icon component.
+ * @constructor
+ */
+const ColorSelectorIcon = ( { style } ) =>
+	<div className="block-editor-block-colors-selector__icon-container">
+		<div
+			className="colors-selector__state-selection wp-block-navigation-menu-item"
+			style={ style }
+		>
+			{ __( 'Aa' ) }
+		</div>
+	</div>;
+
+/**
+ * Renders the Colors Selector Toolbar with the icon button.
+ *
+ * @param {function} onToggle Callback open/close Dropdown.
+ * @param {bool} isOpen True is the color settings dropdown is open. Otherwise, False.
+ * @return {*} React toggle button component.
+ */
+const renderToggle = ( style ) => ( { onToggle, isOpen } ) => {
+	const openOnArrowDown = ( event ) => {
+		if ( ! isOpen && event.keyCode === DOWN ) {
+			event.preventDefault();
+			event.stopPropagation();
+			onToggle();
```

Comment:
Minor: `onToggle` is could be said to be slightly ambiguous. If I call this method in `renderToggle` how do I know it will toggle to "open"? The answer is that we've checked  `! isOpen` above. However, it might be clearer to just have dedicated `toggleOpen` toggleClose` methods?

Reply:
The thing is this `onToggle()` function is given by the function bound to the `renderToggle` property of the `<Dropdown />` component. [README.md](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/dropdown).